person.rb

```diff
@@ -20,7 +27,35 @@ def of_age?
   end
 end
 
-person = Person.new(17)
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    trimmed_name = @nameable.correct_name[0..9]
+    trimmed_name.length == 10 ? trimmed_name : @nameable.correct_name
+  end
+end
```

Comment:
- [ ] [OPTIONAL] It would be nice if all these classes are in a separate file, it will make your code base more readable